### ISSUE TITLE
Huge stack trace dump on pulumi up

### ISSUE THREAD

**Issue by orcutt989** [pulumi/pulumi#11162]
### What happened?

I updated to pulumi 3.44.1 and now periodically I get these massive stack dumps when applying an update.

### Steps to reproduce

`pulumi up -y` will **sometimes** result in this massive stack trace dump and leave an update running

### Expected Behavior

No trace dump and cancelled update on error, or sucessful update.

### Actual Behavior

Stack trace (572 KB too big for github/pastepin)

https://temp-bucket-uvkzyg9kozf1td3srocs.s3.amazonaws.com/trace.txt

### Output of `pulumi about`

```
 pulumi about
CLI
Version      3.44.1
Go Version   go1.19.2
Go Compiler  gc

Plugins
NAME        VERSION
aws         4.21.1
docker      3.1.0
eks         0.33.0
kubernetes  3.7.2
nodejs      unknown

Host
OS       darwin
Version  12.6
Arch     x86_64

This project is written in nodejs: executable='/usr/local/bin/node' version='v18.10.0'

...

Backend
Name           pulumi.com
URL            https://app.pulumi.com/user
User           user
Organizations  user, organization

Dependencies:
NAME            VERSION
@pulumi/awsx    0.31.0
@pulumi/eks     0.33.0
@pulumi/pulumi  3.12.0
@types/node     16.9.6
dotenv          10.0.0
tezos-pulumi    1.0.0
yaml            1.10.2
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by jaxxstorm** [comment]
This is resolved in 3.44.2. Please update your CLI version

**Comment by orcutt989** [comment]
Thank you! I just saw https://github.com/pulumi/pulumi/issues/11153
